### ISSUE TITLE
load when calling hocon file

### ISSUE THREAD

**Issue by dwsmith1983** [zifeo/dataconf#5]
In your parsing operation, you dont allow for optional parameters that are list or dicts.

```python
    if origin is list:
        if len(args) != 1:
            raise MissingTypeException("excepted list with type information: List[?]")
        return [__parse(v, args[0], f"{path}[]") for v in value]

    if origin is dict:
        if len(args) != 2:
            raise MissingTypeException(
                "excepted dict with type information: Dict[?, ?]"
            )
        return {k: __parse(v, args[1], f"{path}.{k}") for k, v in value.items()}
```

Consider the following dataclass:
```python
        @dataclass
        class Base:
            data_root: Text
            pipeline_name: Text
            data_type: Text
            production: bool
            conn: Optional[Conn] = None
            data_split: Optional[Dict[Text, int]] = None
            tfx_root: Optional[Text] = None
            metadata_root: Optional[Text] = None
            beam_args: Optional[List[Text]] = field(
                default_factory=lambda: ["--direct_running_mode=multi_processing", "--direct_num_workers=0"]
            )
```

An optional parameter with None shouldn't return a failure. We use HOCON configs all the time with case classes in JVM languages and optional parameter parsing should be able to be by passed if None is specified.

**Comment by zifeo** [comment]
Closed by #7

**Comment by dwsmith1983** [comment]
@zifeo how can I contact you? I am working on implementation that will replicate a Scala sealed trait being used in the config to add the dataconf. It will add to the flexibility of the library. Do you have a Discord channel? Can you open discussion on repo? Or some other method.
